src/dlgConnectionProfiles.cpp

```diff
@@ -2337,3 +2342,138 @@ QIcon dlgConnectionProfiles::customIcon(const QString& text) const
+        case Qt::Key_B:
+        case Qt::Key_C:
+        case Qt::Key_D:
+        case Qt::Key_E:
+        case Qt::Key_F:
+        case Qt::Key_G:
+        case Qt::Key_H:
+        case Qt::Key_I:
+        case Qt::Key_J:
+        case Qt::Key_K:
+        case Qt::Key_L:
+        case Qt::Key_M:
+        case Qt::Key_N:
+        case Qt::Key_O:
+        case Qt::Key_P:
+        case Qt::Key_Q:
+        case Qt::Key_R:
+        case Qt::Key_S:
+        case Qt::Key_T:
+        case Qt::Key_U:
+        case Qt::Key_V:
+        case Qt::Key_W:
+        case Qt::Key_X:
+        case Qt::Key_Y:
+        case Qt::Key_Z:
+            if (keyEvent->modifiers() & ~(Qt::ShiftModifier)) {
+                // There is a modifier in play OTHER than the shift one so treat
+                // it as normal:
+                return QObject::eventFilter(obj, event);
+            }
+
+            addLetterToProfileSearch(keyEvent->key());
+            // Eat (filter) this event so it goes no further:
+            return true;
+        }
+    }
+
+    // standard event processing
+    return QObject::eventFilter(obj, event);
+}
+
+void dlgConnectionProfiles::addLetterToProfileSearch(const int key)
+{
+    if ((key < 0) || (key > 128)) {
+        // out of range of normal ASCII keys
+        return;
+    }
+
+    // As it happens the values for key correspond to those of the corresponding
+    // ASCII (upper-case for letters) character codes
+    searchText.append(QLatin1Char(static_cast<unsigned char>(key)));
+    auto indexes = findProfilesBeginningWith(searchText);
+
+    if (indexes.isEmpty()) {
+        // No matches at all so clearing search term and reset all profiles to
+        // be enabled:
+        searchText.clear();
+        for (int i = 0, total = profiles_tree_widget->count(); i < total; ++i) {
+            profiles_tree_widget->item(i)->setFlags(profiles_tree_widget->item(i)->flags() | Qt::ItemIsEnabled);
+        }
+        notificationArea->hide();
+        notificationAreaIconLabelWarning->hide();
+        notificationAreaIconLabelError->hide();
+        notificationAreaIconLabelInformation->hide();
+        notificationAreaMessageBox->clear();
+        return;
+    }
+
+    for (int i = 0, total = profiles_tree_widget->count(); i < total; ++i) {
+        auto flags = profiles_tree_widget->item(i)->flags();
+        if (indexes.isEmpty() || !indexes.contains(i)) {
+            flags &= ~Qt::ItemIsEnabled;
+        } else {
+            flags |= Qt::ItemIsEnabled;
+        }
+        profiles_tree_widget->item(i)->setFlags(flags);
+    }
+
+    profiles_tree_widget->setCurrentRow(indexes.first());
+    notificationArea->show();
+    notificationAreaIconLabelWarning->hide();
+    notificationAreaIconLabelError->hide();
+    notificationAreaIconLabelInformation->show();
+    notificationAreaMessageBox->show();
+    notificationAreaMessageBox->setText(tr("As you type more letters in the name of a profile all the non-matching ones "
```

Comment:
Yeah, but since I am actually disabling non-matching profiles we ought to let the end user know how to get them back again. I suppose I could try something more fancy that would only show this message *once* per Mudlet run - but I do not really want to add too much more :spaghetti: code here.

Also - I cannot view streamable.com URLs - they must be (like GitHub) using features that are only compatible with the most bleeding edge browsers and unlike GH (https://github.com/JustOff/github-wc-polyfill) I do not have a fix for them.

Reply:
Otherwise - is this any good? :worried:

Reply:
The message does not need to be shown at all. Good UX does not require explanation!